src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
+
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herbivore) {
+    if (herbivore.hasOwnProperty('hidden') && !herbivore.hidden) {
```

Comment:
Replace the 'hasOwnProperty' check with an instanceof check to make sure the passed object is an instance of Herbivore. For example: 'if (herbivore instanceof Herbivore && !herbivore.hidden)'.